src/metabase/public_settings.clj

```diff
@@ -439,3 +442,29 @@
   :visibility :public
   :type       :integer
   :default    180)
+
+(defsetting cloud-gateway-ips-url
+  "Store URL for fetching the list of Cloud gateway IP addresses"
+  :visibility :internal
+  :setter     :none
+  :default    (str premium-features/store-url "/static/cloud_gateways.json"))
+
+(def ^:private fetch-cloud-gateway-ips-fn
+  (memoize/ttl
+   (fn []
+       (when (premium-features/is-hosted?)
+         (try
+           (-> (http/get (cloud-gateway-ips-url))
+               :body
+               (json/parse-string keyword)
+               :ip_addresses)
+           (catch clojure.lang.ExceptionInfo e
```

Comment:
I think this exception class is not a good choice:
```clojure
  (try
    (-> (http/get #_(cloud-gateway-ips-url)
                  "https://storeee.metabase.com/static/cloud_gateways.json")
        :body
        (json/parse-string keyword)
        :ip_addresses)
    (catch clojure.lang.ExceptionInfo e
      :bummer))
Execution error (UnknownHostException) at java.net.Inet6AddressImpl/lookupAllHostAddr (Inet6AddressImpl.java:-2).
storeee.metabase.com: nodename nor servname provided, or not known
```

It doesn't catch all errors. 

```clojure
public-settings=> (type *e)
java.net.UnknownHostException
public-settings=> (ancestors (type *e))
#{java.lang.Exception java.io.Serializable java.lang.Throwable java.lang.Object java.io.IOException}
```
I think we want to catch `Exception` or perhaps `Throwable`. But a clojure specific error class isn't a great choice for interop with so much java like http requests.

Reply:
I based this off of similar code in `metabase.public-settings.premium-features` that fetches token features from a store endpoint: https://github.com/metabase/metabase/blob/master/src/metabase/public_settings/premium_features.clj#L74

I can definitely change to `Exception` but I'm curious why `clojure.lang.ExceptionInfo` was used there. Maybe that needs to be changed as well.

Reply:
cc @camsaul, seems like the exception caught in `fetch-token-status*` was changed from `Throwable` to `clojure.lang.ExceptionInfo` this PR: https://github.com/metabase/metabase/pull/13590/files#diff-1b2e7bb52ebe54644c80a90c2ba72ffb66d43ef4a1d042c7f587d6b7b49557dcR76

Was there a reason for this?